pkg/meta/tkv_bak.go

```diff
@@ -0,0 +1,850 @@
+
+		start, end := make([]byte, 9), make([]byte, 9)
+		start[0], end[0] = 'A', 'A'
+		m.encodeInode(Ino(left), start[1:])
+		m.encodeInode(Ino(right), end[1:])
+		eg.Go(func() error {
+			return m.execStmt(egCtx, s.txn, func(tx *kvTxn) error {
+				var ent *entry
+				tx.scan(start, end, false, func(k, v []byte) bool {
+					if egCtx.Err() != nil {
+						return false
+					}
+					if len(k) <= 9 || k[0] != 'A' {
+						return true
+					}
+					ino := m.decodeInode(k[1:9])
+					switch k[9] {
+					case 'I':
+						node := s.pools[0].Get().(*pb.Node)
+						node.Inode = uint64(ino)
+						UnmarshalNodePB(v, node)
+						sums[SegTypeNode].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeNode, node
+						entryCh <- ent
+					case 'D':
+						edge := s.pools[1].Get().(*pb.Edge)
+						edge.Parent = uint64(ino)
+						edge.Name = k[10:]
+						typ, inode := m.parseEntry(v)
+						edge.Type, edge.Inode = uint32(typ), uint64(inode)
+						sums[SegTypeEdge].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeEdge, edge
+						entryCh <- ent
+					case 'C':
+						n := len(v) / sliceBytes
+						chk := s.pools[2].Get().(*pb.Chunk)
+						chk.Inode = uint64(ino)
+						chk.Index = binary.BigEndian.Uint32(k[10:])
+						chk.Slices = make([]*pb.Slice, 0, n)
+						var ps *pb.Slice
+						for i := 0; i < n; i++ {
+							ps = s.pools[3].Get().(*pb.Slice)
+							UnmarshalSlicePB(v[i*sliceBytes:], ps)
+							chk.Slices = append(chk.Slices, ps)
+						}
+						sums[SegTypeChunk].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeChunk, chk
+						entryCh <- ent
+					case 'S':
+						sym := s.pools[4].Get().(*pb.Symlink)
+						sym.Inode = uint64(ino)
+						sym.Target = unescape(string(v))
+						sums[SegTypeSymlink].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeSymlink, sym
+						entryCh <- ent
+					case 'X':
+						xattr := s.pools[5].Get().(*pb.Xattr)
+						xattr.Inode = uint64(ino)
+						xattr.Name = string(k[10:])
+						xattr.Value = v
+						sums[SegTypeXattr].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeXattr, xattr
+						entryCh <- ent
+					case 'P':
+						parent := s.pools[6].Get().(*pb.Parent)
+						parent.Inode = uint64(ino)
+						parent.Parent = uint64(m.decodeInode(k[10:]))
+						parent.Cnt = parseCounter(v)
+						sums[SegTypeParent].Add(1)
+						ent = entryPool.Get().(*entry)
+						ent.typ, ent.elem = SegTypeParent, parent
+						entryCh <- ent
+					}
+					return true
+				})
+				return nil
+			})
+		})
+	}
+
+	if iErr := eg.Wait(); iErr != nil {
+		ctx.Cancel()
+		wg.Wait()
+		return iErr
+	}
+
+	close(entryCh)
+	wg.Wait()
+
+	logger.Debugf("dump %s num: %s", s, printSums(sums))
+	return err
+}
+
+func (s *kvMixDS) release(msg proto.Message) {
+	switch list := msg.(type) {
+	case *pb.NodeList:
+		for _, node := range list.List {
+			s.pools[0].Put(node)
```

Comment:
reset fields of node before putting back.

Reply:
done